test/e2e/prioritize.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
+)
+
+var _ = Describe("Prioritize E2E Test", func() {
+	It("Node Affinity Test", func() {
+		context := initTestContext()
+		defer cleanupTestContext(context)
+
+		var preferredSchedulingTermSlice []v1.PreferredSchedulingTerm
+		nodeSelectorRequirement := v1.NodeSelectorRequirement{Key: "kubernetes.io/hostname", Operator: v1.NodeSelectorOpIn, Values: []string{"kube-node-2"}}
```

Comment:
get node from node list instead of hardcode; we also need to support different environment, e.g. `./hack/localup.sh`